python/api/src/araali/aws.py

```diff
@@ -21,6 +21,38 @@ def cf_rm(name):
 def cf_add_vm():
     print("adding vm")
 
+def cf_launch_fortified_vm(stack_name, email, key_pair):
+    template_url = "https://s3.us-west-1.amazonaws.com/araalinetworks.test/quickstart_vm/AraaliVMQuickstartStack.template.json"
+    rc = utils.run_command("aws cloudformation create-stack --stack-name %s --template-url %s\
+        --parameters ParameterKey=KeyName,ParameterValue=%s ParameterKey=email,ParameterValue=%s" % (
+        stack_name, template_url, key_pair, email), debug=False, result=True, strip=False)
+    if rc[0] != 0:
+        print("*** failed: %s" % rc[1].decode())
+        return False, None
+    return True, rc[1]
+
+def cf_launch_eks_cluster(stack_name, cluster_name, availability_zones):
+    template_url = "https://s3.amazonaws.com/aws-quickstart/quickstart-amazon-eks/templates/amazon-eks-entrypoint-new-vpc.template.yaml"
+    availability_zone_str = ""
+    for zone in availability_zones:
+        availability_zone_str += "\\\\," + zone
+    availability_zone_str = availability_zone_str.strip("\\,")
+    rc = utils.run_command("aws cloudformation create-stack --stack-name %s --template-url %s --parameters ParameterKey=AvailabilityZones,ParameterValue=%s\
+     ParameterKey=EKSClusterName,ParameterValue=%s ParameterKey=NumberOfAZs,ParameterValue=2 --capabilities CAPABILITY_IAM CAPABILITY_AUTO_EXPAND" % (
+        stack_name, template_url, availability_zone_str, cluster_name), debug=False, result=True, strip=False)
+    if rc[0] != 0:
+        print("*** failed: %s" % rc[1].decode())
```

Comment:
if this is a pattern you like, move the print inside run_commmands(). Then your return code can be simpler, something like:

`return run_command("blah")`

Or if you dont want to touch the base function, wrap it up and create a new one: run_command_logfailure()